### PR TITLE
Bump the notary version to one that fixes a bug with delegation path traversal

### PR DESCRIPTION
**Pull request by cyli**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -249,3 +249,5 @@ ENTRYPOINT ["hack/dind"]
 
 # Upload docker source
 COPY . /go/src/github.com/docker/docker
+
+RUN set -x && GOPATH=$(pwd)/vendor/src/github.com/docker/notary/Godeps/_workspace:$GOPATH go build -o /usr/local/bin/notary-server github.com/docker/notary/cmd/notary-server &&GOPATH=$(pwd)/vendor/src/github.com/docker/notary/Godeps/_workspace:$GOPATH go build -o /usr/local/bin/notary github.com/docker/notary/cmd/notary
```
Comment (icecrime) [inline Dockerfile]: Can you please clarify why that's needed? Thanks

Comment (cyli) [inline Dockerfile]: Oh sorry, that was a bug in my autovendoring notary script.  :)  Fixing

**Comment by tiborvass** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

**Pull request merged**